cds_ils/importer/documents/importer.py

```diff
@@ -257,6 +257,29 @@ def _validate_match_identifiers(self, existing_document):
         # not valid - will create a new document
         return False
 
+    def _validate_volumes(self, existing_document):
+        """Validate serial volumes match."""
+        import_doc_serial = self.json_data.get("_serial", None)
+        existing_doc_serials = existing_document.relations.get('serial', [])
+        if import_doc_serial and existing_doc_serials:
+            import_volume = import_doc_serial.get("volume")
+            import_serial_title = import_doc_serial["title"].lower()
+
+            same_serial, same_volume, both_have_volumes = False, False, False
+
+            for serial in existing_doc_serials:
+                existing_volume = serial.get["volume"]
+                existing_title = serial["record_metadata"]["title"].lower()
+                if existing_title == import_serial_title:
+                    same_serial = True
+                    both_have_volumes = all([import_volume, existing_volume])
+                    same_volume = existing_volume == import_volume
+            if same_serial and both_have_volumes and same_volume:
+                return True
+            return False
+        else:
+            return True
```

Comment:
Very minor and just IMHO, feel free to skip this.
I would move this `return True` more up to improve readability maybe? Even if it is considered not a very good practice to have multiple `return`...

```
if not (if import_doc_serial and existing_doc_serials):
    return True

....
```